### ISSUE TITLE
Can't connect devices

### ISSUE THREAD

**Issue by alecsito** [Ubiquiti-App/UNMS#81]
Hello everyone, im from Argentina. 
I try to explain my problem, currently i have unms version 0.10.2 and i can't connect my devices. When i try to connec the system returns me the following error "Connection attempt has failed (timeout error)" .  
UNMS  it's installed on the ip 192.168.2.111 and use the same ip  for "Host" and "Ip Address". Also probe in diferents PC but the result is the same. The version of the devices is  v6.0.7 (XM)
Recently i use this guide https://github.com/Ubiquiti-App/UNMS/wiki/Register-Devices-to-UNMS but the problem persist.
I would appreciate some help , thanks you very much.

**Comment by jflidr** [comment]
Hi @alecsito it looks like that your device can't connect back to UNMS. Could you please check this [Discovery and connection tutorial](https://github.com/Ubiquiti-App/UNMS/wiki/Discovery-Troubleshooting#i-can-discover-the-device-but-connection-to-unms-is-failing) and your [devices logs](https://github.com/Ubiquiti-App/UNMS/wiki/Discovery-Troubleshooting#where-to-find-logs-on-the-device). You can check these [typical issues](https://github.com/Ubiquiti-App/UNMS/wiki/Device-log-Troubleshooting).

**Comment by oreo9810** [comment]
@jflidr I appreciate the links and information you posted above.  I am having a similar issue.  I am new to UNMS and I just installed and configured it yesterday.  Today I attempted to connect two devices but they will not connect.  I can discover them but they error out when attempting to connect: "Server request failed: ECONNREFUSED".  I SSH connected to the endpoints and I can ping the UNMS server, but I cannot curl.  When I type the curl command suggested in the troubleshooting page you linked to, I receive a error: "-sh: curl: not found".  That suggests to me that the command is not found.  I also configured the URL and Key on the devices and that did not change the connection status in UNMS.  It did, however, prompt an error in the UNMS logs: "Problem with establishing connection from 10.x.x.x.  Check the UNMS Key on the device."  I have confirmed the key on both the UNMS server and the endpoints.  Any suggestions would be greatly appreciated.

**Comment by MajoV** [comment]
A similar problem has been discussed here
[[]](https://community.ubnt.com/t5/UNMS-Ubiquiti-Network-Management/Server-request-failed-ECONNREFUSED/m-p/2025136#M537)

**Comment by oreo9810** [comment]
@MajoV Thanks for the link.  I was using custom ports and that was causing the ECONNREFUSED error.  After changing the ports I was able attempt to connect to the devices.  However, now I am receiving a time out error when I attempt to connect the devices through the UNMS discovery page.

**Comment by oreo9810** [comment]
Update: I was able to get my devices to connect to UNMS.  I had a network/dns problem that prevented the connection via the DNS name.  I found this out by confirming I could not ping using the DNS name, but I could ping using the local IP address of the UNMS server.  Once that was fixed I did have to remove the device/mac from the AES database on the UNMS server as outlined in the post here: https://github.com/Ubiquiti-App/UNMS/wiki/Device-log-Troubleshooting#problem-with-aes-encryption

Thanks @MajoV for pointing me in the right direction!
